site-src/geps/gep-851.md

```diff
@@ -0,0 +1,102 @@
+# GEP-851: Allow Multiple Certificate Refs per Gateway Listener
+
+* Issue: [#851](https://github.com/kubernetes-sigs/gateway-api/issues/851)
+* Status: Implementable
+
+## TLDR
+
+Replace `CertificateRef` field with an `CertificateRefs` field in Gateway
+Listeners.
+
+## Goals
+
+* Allow certs to be attached for multiple unique Hostnames for the same
+  Listener. This is most helpful in the following situations:
+  * The hostname on the Listener is empty or unspecified allowing Routes to
+    attach arbitrary hostnames to the Listener with Routes.
+  * The hostname on the Listener is a wildcard and Route owners want to attach
+    multi-level subdomains. For example, `*.example.com` on the Listener and
+    `app.dev.example.com` on a Route.
```

Comment:
Please see this thread: https://github.com/kubernetes-sigs/gateway-api/pull/856/files#r705706904.
I don't think multi-level subdomains are possible with a wildcard hostname in Listener.

Reply:
Followed up in that thread.

Reply:
I read this as meaning that the ability to provide more than one certificate would allow you to provision different certificates for the two names. We can just make that explicit in the text though.